src/spring-cloud/azext_spring_cloud/tanzu.py

```diff
@@ -400,6 +406,15 @@ def _set_pattern_for_deployment(patterns):
     }
 
 
+def _update_deployment_pattern(client, resource_group, service, name, deployment, config_file_patterns):
+    deployment_properties = deployment.properties
+    deployment_properties.deployment_settings.addon_config = _set_pattern_for_deployment(config_file_patterns)
+    deployment_poller = client.deployments.create_or_update(resource_group, service, name, deployment.name,
+                                                            properties=deployment_properties, sku=deployment.sku)
+    while not _poller_finished(deployment_poller):

```

Comment:
Consider using [`LongRunningOperation`](https://github.com/VSChina/azure-cli-extensions/blob/4fd8cf46508c33bde609f8e4ff7bd8066835c5ca/src/spring-cloud/azext_spring_cloud/tanzu.py#L262) to wait the poller